File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/wallet/DeleteWalletTest.java
package org.hyperledger.indy.sdk.wallet;

import org.hyperledger.indy.sdk.IndyIntegrationTest;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertNotNull;

import org.hyperledger.indy.sdk.InvalidStateException;
import org.json.JSONObject;
import org.junit.Test;

import java.util.concurrent.ExecutionException;


public class DeleteWalletTest extends IndyIntegrationTest {

	@Test
	public void testDeleteWalletWorks() throws Exception {
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		Wallet.deleteWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		Wallet.deleteWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
	}

	@Test
	public void testDeleteWalletWorksForClosed() throws Exception {
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();

		Wallet wallet = Wallet.openWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		assertNotNull(wallet);

		wallet.closeWallet().get();
		Wallet.deleteWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		Wallet.deleteWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
	}

	@Test
	public void testDeleteWalletWorksForOpened() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStateException.class));

		String config = new JSONObject()
				.put("id", "deleteWalletWorksForOpened")
				.toString();

		Wallet.createWallet(config, WALLET_CREDENTIALS).get();
		Wallet.openWallet(config, WALLET_CREDENTIALS).get();
		Wallet.deleteWallet(config, WALLET_CREDENTIALS).get();
	}

	@Test
	public void testDeleteWalletWorksForTwice() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletNotFoundException.class));

		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();

		Wallet wallet = Wallet.openWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();

		wallet.closeWallet().get();

		Wallet.deleteWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		Wallet.deleteWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
	}

	@Test
	public void testDeleteWalletWorksForNotCreated() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletNotFoundException.class));

		Wallet.deleteWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/WalletTests/DeleteWalletTest.cs
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.WalletTests
{
    [TestClass]
    public class DeleteWalletTest : IndyIntegrationTestBase
    {
        [TestMethod]
        public async Task TestDeleteWalletWorks()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            await Wallet.DeleteWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            await Wallet.DeleteWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
        }

        [TestMethod]
        public async Task TestDeleteWalletWorksForClosed()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            var wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            Assert.IsNotNull(wallet);

            await wallet.CloseAsync();
            await Wallet.DeleteWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            await Wallet.DeleteWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
        }

        [TestMethod]
        public async Task TestDeleteWalletWorksForOpened()
        {
            var config = JsonConvert.SerializeObject(new { id = "deleteWalletWorksForOpened" });

            await Wallet.CreateWalletAsync(config, WALLET_CREDENTIALS);
            var wallet = await Wallet.OpenWalletAsync(config, WALLET_CREDENTIALS);

            var ex = await Assert.ThrowsExceptionAsync<InvalidStateException>(() =>
                Wallet.DeleteWalletAsync(config, WALLET_CREDENTIALS)
            );

            await wallet.CloseAsync();
        }

        [TestMethod]
        public async Task TestDeleteWalletWorksForTwice()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            var wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            await wallet.CloseAsync();

            await Wallet.DeleteWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);

            var ex = await Assert.ThrowsExceptionAsync<WalletNotFoundException>(() =>
                 Wallet.DeleteWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS)
            );        
        }

        [TestMethod]
        public async Task TestDeleteWalletWorksForNotCreated()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletNotFoundException>(() =>
                Wallet.DeleteWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS)
            );
        }
    }
}
